src/main/java/com/shipmentEvents/handlers/EventHandler.java

```diff
@@ -1,195 +1,195 @@
+    
+            //Create a new file in the Constants.SUMMARY_BUCKET
+            logger.log("Map of statuses -> " + latestStatusForTrackingNumber);
+            String summaryUpdateName = Long.toString(System.currentTimeMillis());
             
-            if (!fileContents.contains("\n")) {
-                
+            EventHandler.getS3Client().putObject(Constants.SUMMARY_BUCKET, summaryUpdateName, latestStatusForTrackingNumber.toString());
+            
+            long expirationTime = System.currentTimeMillis() + Duration.ofMinutes(1).toMillis();
+            while(System.currentTimeMillis() < expirationTime) {
+                if (s3Client.doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) {
+                    break;
+                }
+                logger.log("waiting for file to be created " + summaryUpdateName);
+                Thread.sleep(1000);
             }
-            String[] lines = fileContents.split("\n");
-            String line1 = lines[0];
-            String line2 = lines[1];
-
-            String status = line1.split(":")[1];
-            Long timeStamp = Long.parseLong(line2.split(":")[1]);
-
-
-            if (null == lastKnownStatus || lastKnownStatus.getLeft() < timeStamp) {
-                lastKnownStatus = new MutablePair<Long, String>(timeStamp, status);
-                latestStatusForTrackingNumber.put(trackingNumber, lastKnownStatus);
+            
+            // Before we delete the shipment updates make sure the summary update file exists
+            if (EventHandler.getS3Client().doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) {
+                deleteProcessedFiles(filesToDelete);
+                logger.log("All updates successfully processed");
+            } else {
+                throw new RuntimeException("Failed to write summary status, will be retried in 15 minutes");
             }
-
-            //Add to list of processed files
-            filesProcessed.add(new KeyVersion(summary.getKey()));
-            logger.log("logging Contents of the file" + fileContents);
+            
         }
-        return filesProcessed;
-    }
     
-
-    private void deleteProcessedFiles(Map<String, List<KeyVersion>> filesToDelete) {
-      final AmazonS3 s3Client = EventHandler.getS3Client();
-      for (Entry<String, List<KeyVersion>> entry : filesToDelete.entrySet()) {
-          final DeleteObjectsRequest deleteRequest = new DeleteObjectsRequest(entry.getKey()).withKeys(entry.getValue()).withQuiet(false);
-          s3Client.deleteObjects(deleteRequest);
-      }
-    }
+        private List<KeyVersion> processEventsInBucket(String bucketName, LambdaLogger logger, Map<String, Pair<Long, String>> latestStatusForTrackingNumber) {
+            final AmazonS3 s3Client = EventHandler.getS3Client();
+            logger.log("Processing Bucket: " + bucketName);
+    
+            ObjectListing files = s3Client.listObjects(bucketName);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses an outdated API. [ListObjectsV2 is the revised List Objects API, and we recommend you use this revised API for new application developments](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html).